File: KHost.App/ClientApp/src/app/models/Download.ts
export class Download {

    name: string;

    songId: number;

    id?: number;

    progress: number = 0;

    constructor({
        name,
        songId,
        id,
        progress
    }: parameters) {
        this.name = name;
        this.songId = songId;
        this.id = id;
        this.progress = progress;
    }
}

interface parameters {
    name: string,
    songId: number,
    id: number,
    progress: number
}

File: KHost.Common/Models/Download.cs
namespace KHost.Common.Models
{
    public class Download : BaseModel, IModelWithId
    {
        public int? Id { get; set; }

        public int SongId { get; set; }

        public string Name { get; set; }

        public int Progress { get; set; }

        public override Download Clone() => base.Clone() as Download;
    }
}
